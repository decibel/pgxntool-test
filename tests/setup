#!/bin/bash

trap 'echo "$BASH_SOURCE: line $LINENO" >&2' ERR
set -o errexit -o errtrace -o pipefail

BASEDIR=`cd ${0%/*}; pwd`

. $BASEDIR/../.env
. $TOPDIR/lib.sh

cd $TEST_REPO

out Making checkout dirty
touch garbage
git add garbage
out Verify setup.sh errors out
if pgxntool/setup.sh; then
  echo "setup.sh should have exited non-zero" >&2
  exit 1
fi
git reset HEAD garbage
rm garbage

out Running setup.sh
pgxntool/setup.sh

out -v Status
ls
git status
git diff


out git commit
git commit -am "Test setup"

check_log

# vi: expandtab sw=2 ts=2

