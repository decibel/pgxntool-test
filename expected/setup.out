# Making checkout dirty
# Verify setup.sh errors out
diff --git a/garbage b/garbage
new file mode 100644
index 0000000..e69de29
Git repository is not clean; please commit and try again.
# Running setup.sh
Copying pgxntool/_.gitignore to .gitignore and adding to git
Copying pgxntool/META.in.json to META.in.json and adding to git
Creating Makefile
make[1]: `META.json' is up to date.
Copying ../pgxntool/test/deps.sql to deps.sql and adding to git
On branch master
Your branch is ahead of 'origin/master' by 2 commits.
  (use "git push" to publish your local commits)
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	new file:   ../.gitignore
	new file:   ../META.in.json
	new file:   ../META.json
	new file:   ../Makefile
	new file:   deps.sql
	new file:   pgxntool

If you won't be creating C code then you can:

rmdir src

If everything looks good then

git commit -am 'Add pgxntool (https://github.com/decibel/pgxntool/tree/release)'
######################################
# Status
######################################
META.in.json
META.json
Makefile
meta.mk
pgxntool
pgxntool-test.control
sql
src
t
test
On branch master
Your branch is ahead of 'origin/master' by 2 commits.
  (use "git push" to publish your local commits)
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	new file:   .gitignore
	new file:   META.in.json
	new file:   META.json
	new file:   Makefile
	new file:   test/deps.sql
	new file:   test/pgxntool

# git commit
@GIT COMMIT@ Test setup
 6 files changed, 259 insertions(+)
 create mode 100644 .gitignore
 create mode 100644 META.in.json
 create mode 100644 META.json
 create mode 100644 Makefile
 create mode 100644 test/deps.sql
 create mode 120000 test/pgxntool
